src/scripts/profiling/run_profiling.py

```diff
@@ -1,53 +1,232 @@
+        "pop_df_times_extended": int(
+            np.ceil((pops.props.shape[0] - pops.initial_size) / pops.new_rows.shape[0])
+        ),
+    }
+
+    return pop_stats
+
+
+def record_disk_statistics(
+    disk_usage: Dict[str, Union[int, float]]
+) -> Dict[str, Union[int, float]]:
+    """
+    Extract disk I/O statistics from the profiled run.
+    Statistics are returned as a dictionary.
+
+    Key / value pairs are:
+    disk_reads: int
+        Number of times the disk was read during the simulation
+    disk_writes: int
+        Number of times the disk was written to during the simulation
+    disk_read_MB: float
+        Memory read in MBs from the disk during simulation
+    disk_write_MB: float
+        Memory written in MBs from the disk during simulation
+    disk_read_s: float
+        Time in seconds spent reading from disk during simulation
+    disk_write_s: float
+        Time in seconds spent writing to disk during simulation
+    """
+    if disk_usage is None:
+        return dict()
+
+    disk_stats = {
+        "disk_reads": disk_usage["read_count"],
+        "disk_writes": disk_usage["write_count"],
+        "disk_read_MB": disk_usage["read_bytes"] / 1e6,
+        "disk_write_MB": disk_usage["write_bytes"] / 1e6,
+        "disk_read_s": disk_usage["read_time"] / 1e3,
+        "disk_write_s": disk_usage["write_time"] / 1e3,
+    }
+
+    return disk_stats
+
+
+def record_profiling_session_statistics(
+    session: Session,
+) -> Dict[str, Union[int, float]]:
+    """
+    Extract important profiling statistics from the session that was captured.
+    Statistics are returned as a dictionary.
+
+                "start_time": self.start_time,
+            "duration": self.duration,
+            "sample_count": self.sample_count,
+            "start_call_stack": self.start_call_stack,
+            "program": self.program,
+            "cpu_time": self.cpu_time,
+
+    Key / value pairs are:
+    start_time: float
+        Time (stored as a float representing the number of seconds from a reference time)
+        the profiling session started.
+    duration: float
+        Number of seconds that the profiling session lasted for.
+    cpu_time: float
+        Number of seconds of CPU time that were used by the program during the profiling
+        session.
+    """
+    if session is None:
+        return dict()
+
+    profiler_stats = {
+        "start_time": session.start_time,
+        "duration": session.duration,
+        "cpu_time": session.cpu_time,
+    }
+
+    return profiler_stats
+
+
+def record_run_statistics(
+    output_file: str,
+    html_output_file: str = None,
+    profiling_session: Session = None,
+    completed_sim: Simulation = None,
+    disk_usage: Dict[str, Union[int, float]] = None,
+    **additional_stats: str,
+) -> None:
+    """
+    Organise all statistics to be collected from the profiling run into a single dict,
+    which can then be dumped to a JSON file.
```

Comment:
```suggestion
    which is then be dumped to a JSON file.
```

`can then be` originally made me think the dumping to JSON file happened elsewhere.